### ISSUE TITLE
Encyption codec verification does not work correctly

### ISSUE THREAD

**Issue by peter-fingo** [tekartik/sembast.dart#28]
It seems like the process for codec verification is this:

1. When the database is opened with the codec, encrypt `{'signature': ${codec.signature}}` with the codec, and store that value.
2. The next time the database is opened, calculate the signature on the passed value and compare to the stored value. If there is a mismatch, stop.

The problem is that this approach only works if the encryption algorithm always produces the same output when given the same input, which is actually a very undesirable property for  an encryption algorithm to have. This rules out any algorithm that uses a random salt, for instance.

Proposed solutions:
- compare the bare text signatures passed, rather than the encrypted versions
- Switch (2) above to the following: decrypt the stored signature with the passed codec and compare to the passed signature.

**Comment by peter-fingo** [comment]
That seems fine to me, but I don't understand what adding the random portion does to help compared to the 2nd solution in my comment.

**Comment by alextekartik** [comment]
I though adding a random seed would add an extra level of security, i.e. you don't know what has been encrypted in the signature. I'm fine without the random number and the format is even backward and foreward compatible

**Comment by peter-fingo** [comment]
New checkCodec function:
```
bool checkCodecSignature(String signature, SembastCodec codec) {
     if (signature == null) return codec == null;
     if (codec == null) return false;
     try {  // if decryption fails, the check should fail.
       final decryptedSignature = codec.codec.decode(signature)['signature'];
       return decryptedSignature == codec.signature;
     } catch (_) {
       return false;
     }
}
```

I don't think I can submit a PR to the repo directly, unless I'm missing something?

**Comment by alextekartik** [comment]
I was already on it. Sorry. That's about that. I added some tests using a codec that does add some random seeds. Will ping you when published.

**Comment by alextekartik** [comment]
Published v 1.15.2. Thanks again for your report.
